engine/src/flutter/lib/web_ui/lib/src/engine/semantics/route.dart

```diff
@@ -6,6 +6,19 @@ import '../dom.dart';
 import '../semantics.dart';
 import '../util.dart';
 
+mixin RouteLike on SemanticRole {
```

Comment:
It seems like all our route types are expressible using a simple class hierarchy:

```
class SemanticRouteBase extends SemanticRole;
class SemanticRoute extends SemanticRouteBase;
class SemanticDialog extends SemanticRouteBase;
class SemanticAlertDialog extends SemanticRouteBase;
```